test/basics_datamunger_test.js

```diff
@@ -391,34 +441,234 @@ describe('basics datamunger', function() {
     });
   });
 
+  describe('getLatestPumpUploaded', function() {
+    it('should be a function', function() {
+      assert.isFunction(dm.getLatestPumpUploaded);
+    });
+
+    it('should return a pump with proper data', function() {
+      var patientData = {
+        grouped: {
+          pumpSettings: [
+            {
+              source: constants.TANDEM,
+            },
+          ],
+        },
+      };
+      expect(dm.getLatestPumpUploaded(patientData)).to.equal(constants.TANDEM);
+    });
+
+    it('should return null without proper data', function() {
+      var patientData = {
+        grouped: {
+          pumpSettings: [],
+        },
+      };
+      expect(dm.getLatestPumpUploaded(patientData)).to.equal(null);
+    });
+  });
+
+  describe('processInfusionSiteHistory', function() {
+    it('should be a function', function() {
+      assert.isFunction(dm.processInfusionSiteHistory);
+    });
+
+    it('should return null without latest pump', function() {
+      var basicsData = {
+        data: {},
+        sections: siteChangeSections,
+      };
+
+      var patient = {
+        permissions: {
+          'root': {},
+        },
+        profile: {
+          fullName: 'Jill Jellyfish',
+          patient: {
+            profile: {
+              settings: {
+                siteChangeSource: constants.SITE_CHANGE_CANNULA,
+              },
+            },
+          },
+        },
+      };
+
+      expect(dm.processInfusionSiteHistory(basicsData, null, patient)).to.equal(null);
+    });
+
+    it('should return upload permission true', function() {
```

Comment:
Optional, but I personally prefer a bit more description around the tests, maybe in additional `describe` blocks if that works best. It's hard to get a sense for what the different circumstances are being testing in this and the next test, example, with the specs just asserting opposites: `should return upload permission true` and `should return upload permission false`.